tests/semantic.at

```diff
@@ -112,6 +112,7 @@ NEGATIVE_TEST([comp_types],[semantic])
 POSITIVE_TEST([comp_opt],[semantic])
 NEGATIVE_TEST([counter2],[semantic])
 POSITIVE_TEST([counter],[semantic])
+POSITIVE_TEST([cmp],[semantic])
```

Comment:
A more descriptive name would be great - `comparator`, or even 'comparator_indirect`.